### PR TITLE
Feature/ep 4015 support first and last processes

### PR DESCRIPTION
**Pull request by JeroenVerstraelen**

### CODE REVIEW

File: openeo-geotrellis/src/test/java/org/openeo/geotrellis/TestOpenEOProcessScriptBuilder.java [hunk 1]
```diff
@@ -694,6 +694,46 @@ public void testIfWithRejectCube() {
...
+        builder.expressionEnd("last",arguments);
+
+        Function1<Seq<Tile>, Seq<Tile>> transformation = builder.generateFunction();
+        ByteArrayTile tile0 = ByteConstantNoDataArrayTile.fill((byte) 10, 4, 4);
+        ByteArrayTile tile1 = ByteConstantNoDataArrayTile.fill((byte) 5, 4, 4);
+        Seq<Tile> result = transformation.apply(JavaConversions.asScalaBuffer(Arrays.asList(tile0, tile1)));
+        Tile res = result.apply(0);
+        assertTileEquals(tile1, res);
+    }
+
```
Comment (soxofaan) [inline openeo-geotrellis/src/test/java/org/openeo/geotrellis/TestOpenEOProcessScriptBuilder.java:736]: Can you also add a test with nodata values in the input and check if the `ignore_nodata` option works properly?

Comment (JeroenVerstraelen) [reply openeo-geotrellis/src/test/java/org/openeo/geotrellis/TestOpenEOProcessScriptBuilder.java:838]: Tests have been added.

**Comment by JeroenVerstraelen** [comment]
- Ensured that the first/last process do not change the cell type to Double
- Increased the performance of the first/last process
- Added extra tests for the ignore_nodata parameter

**Comment by jdries** [comment]
Looks good!

**Pull request merged**